platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/chatter/network/framework/ChatterInstance.java

```diff
@@ -0,0 +1,236 @@
+    public void applyNodeConfig(final NodeConfig nodeConfig) {
+        newEventCreator.applyNodeConfig(nodeConfig);
+        eventPipeline.applyNodeConfigAndCallNext(nodeConfig);
+    }
+
+    /**
+     * Returns a list of messages to gossip. This list may include more than just events.
+     *
+     * @return a list of messages to gossip
+     */
+    public List<GossipMessage> getMessagesToGossip() {
+        final List<GossipMessage> gossipMessages = new ArrayList<>();
+        for (long peerId : peerIds) {
+            if (peerId == selfId.getId()) {
+                continue;
+            }
+            final PeerInstance peer = core.getPeerInstance(peerId);
+
+            SelfSerializable message = peer.outputAggregator().getMessage();
+            while (message != null) {
+                gossipMessages.add(GossipMessage.toPeer(message, selfId.getId(), peerId));
+                message = peer.outputAggregator().getMessage();
+            }
+        }
+        return gossipMessages;
+    }
+
+    /**
+     * The chatter entry point. Messages are received from the gossip simulator and sent here.
+     *
+     * @param msg      the message received
+     * @param fromPeer the peer who sent the message
+     */
+    @Override
+    public void handleMessageFromWire(final SelfSerializable msg, final long fromPeer) {
+        try {
+            if (msg instanceof final SimulatedChatterEvent event) {
+                // Create a copy so that each node sets its own time received
+                final SimulatedChatterEvent eventCopy = event.copy();
+                eventCopy.setTimeReceived(time.now());
+                core.getPeerInstance(fromPeer).inputHandler().handleMessage(eventCopy);
+            } else {
+                core.getPeerInstance(fromPeer).inputHandler().handleMessage(msg);
+            }
+        } catch (PeerMessageException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Prints the current state of the chatter instance and event pipeline. Useful for debugging.
+     */
+    public void printResults() {
+        eventPipeline.printCurrentStateAndCallNext();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public NodeId getNodeId() {
+        return selfId;
+    }
+
+    private static Metrics newDefaultMetrics(final NodeId selfId) {
+        final MetricKeyRegistry registry = mock(MetricKeyRegistry.class);
+        when(registry.register(any(), any(), any())).thenReturn(true);
+        final Configuration configuration = ConfigurationBuilder.create()
+                .withConfigDataType(MetricsConfig.class)
+                .build();
+        final MetricsConfig metricsConfig = configuration.getConfigData(MetricsConfig.class);
+        return new DefaultMetrics(
+                selfId,
+                registry,
+                Executors.newSingleThreadScheduledExecutor(),
+                new DefaultMetricsFactory(),
+                metricsConfig);
```

Comment:
Are you actually using the metrics reported during this run? If not, `NoOpMetrics` might be simpler.

Reply:
Metrics could be used to make assertions in the future, but none are being made now. I changed it to `NoOpMetrics`